lib/exq/mock.ex

```diff
@@ -0,0 +1,175 @@
+  """
+  def set_mode(mode) when mode in [:redis, :inline, :fake] do
+    GenServer.call(__MODULE__, {:mode, self(), mode}, @timeout)
+  end
+
+  @doc """
+  List of enqueued jobs
+
+  This only works if the mode is set to `:fake`
+  """
+  def jobs do
+    GenServer.call(__MODULE__, {:jobs, self()}, @timeout)
+  end
+
+  ### Private
+
+  @impl true
+  def init(options) do
+    {:ok, %State{default_mode: Keyword.get(options, :mode, :redis)}}
+  end
+
+  @doc false
+  def enqueue(pid, queue, worker, args, options) do
+    {:ok, runnable} =
+      GenServer.call(
+        __MODULE__,
+        {:enqueue, self(), :enqueue, [pid, queue, worker, args, options]},
+        @timeout
+      )
+
+    runnable.()
+  end
+
+  @doc false
+  def enqueue_at(pid, queue, time, worker, args, options) do
+    {:ok, runnable} =
+      GenServer.call(
+        __MODULE__,
+        {:enqueue, self(), :enqueue_at, [pid, queue, time, worker, args, options]},
+        @timeout
+      )
+
+    runnable.()
+  end
+
+  @doc false
+  def enqueue_in(pid, queue, offset, worker, args, options) do
+    {:ok, runnable} =
+      GenServer.call(
+        __MODULE__,
+        {:enqueue, self(), :enqueue_in, [pid, queue, offset, worker, args, options]},
+        @timeout
+      )
+
+    runnable.()
+  end
+
+  @impl true
+  def handle_call({:enqueue, owner_pid, type, args}, _from, state) do
+    state = maybe_add_and_monitor_pid(state, owner_pid, state.default_mode)
+
+    case state.modes[owner_pid] do
+      :redis ->
+        runnable = fn -> apply(Redis, type, args) end
+        {:reply, {:ok, runnable}, state}
+
+      :inline ->
+        runnable = fn ->
+          job = to_job(args)
+          apply(job.class, :perform, job.args)
+          {:ok, job.jid}
+        end
+
+        {:reply, {:ok, runnable}, state}
+
+      :fake ->
+        job = to_job(args)
+        state = update_in(state.jobs[owner_pid], &((&1 || []) ++ [job]))
+
+        runnable = fn ->
+          {:ok, job.jid}
+        end
+
+        {:reply, {:ok, runnable}, state}
+    end
+  end
+
+  def handle_call({:mode, owner_pid, mode}, _from, state) do
+    state = maybe_add_and_monitor_pid(state, owner_pid, mode)
+    {:reply, :ok, state}
+  end
+
+  def handle_call({:jobs, owner_pid}, _from, state) do
+    jobs = state.jobs[owner_pid] || []
+    {:reply, jobs, state}
+  end
+
+  @impl true
+  def handle_info({:DOWN, _, _, pid, _}, state) do
+    {_, state} = pop_in(state.modes[pid])
+    {_, state} = pop_in(state.jobs[pid])
+    {:noreply, state}
+  end
+
+  defp to_job([_pid, queue, worker, args, _options]) do
+    %Job{
+      jid: UUID.uuid4(),
+      queue: queue,
+      class: worker,
+      args: args,
+      enqueued_at: DateTime.utc_now()
+    }
+  end
+
+  defp to_job([_pid, queue, _time_or_offset, worker, args, _options]) do
+    %Job{
+      jid: UUID.uuid4(),
+      queue: queue,
+      class: worker,
+      args: args,
+      enqueued_at: DateTime.utc_now()
```

Comment:
Why did you decide not to use time_or_offset and set DateTime.utc_now() for Mock?

Reply:
I can't think of any specific reason. What your use case? do you depend on the timestamp for enqueue_at or enqueue_in in your tests? I am open to fix this if there are use cases I didn't consider.

Reply:
Yes, we need to check is "enqueued_at" equal Date Time that we set

Reply:
fixed by https://github.com/akira/exq/pull/449